src/main/scala/com/monsanto/arch/cloudformation/model/resource/EC2.scala

```diff
@@ -232,6 +232,21 @@ object CidrBlock extends DefaultJsonProtocol {
   }
 }
 
+case class IPAddress(a: IPAddressSegment, b: IPAddressSegment, c: IPAddressSegment, d: IPAddressSegment) {
+  def toJsString: JsString =  JsString( Seq(a, b, c, d).map(_.value.toString).mkString("."))
```

Comment:
It's not clear to me why `IpAddress` has a `toJsString` method. 

Why not just `write` it as a `JsString` in its `def write(obj: IPAddress)` method?